name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/columns/DividendPaymentColumn.java

```diff
@@ -0,0 +1,165 @@
+    {
+        Column column = new Column("nextdivpmtdate", Messages.ColumnDividendsNextPaymentDate, SWT.LEFT, 80); //$NON-NLS-1$
+        column.setMenuLabel(Messages.ColumnDividendsNextPaymentDate_MenuLabel);
+        column.setGroupLabel(Messages.GroupLabelDividends);
+        column.setVisible(false);
+
+        LocalDate now = LocalDate.now();
+
+        Function<Object, LocalDate> dataProvider = element -> {
+            Security sec = Adaptor.adapt(Security.class, element);
+            if (sec == null)
+            {
+                return null; //
+            }
+            return sec.getEvents().stream() //
+                            .filter(DividendEvent.class::isInstance) //
+                            .map(e -> ((DividendEvent) e).getPaymentDate()) //
+                            .sorted(Comparable::compareTo) //
+                            .filter(d -> !now.isAfter(d)) //
+                            .findFirst().orElse(null);
+        };
+
+        column.setLabelProvider(new DateLabelProvider(dataProvider)
+        {
+            @Override
+            public Color getBackground(Object element)
+            {
+                LocalDate date = dataProvider.apply(element);
+                if (date == null || !date.equals(LocalDate.now()))
+                {
+                    return null; //
+                }
+
+                return Colors.theme().greenBackground();
+            }
+        });
+        column.setSorter(ColumnViewerSorter.create(dataProvider::apply));
+        return column;
+    }
+
+    static Column createNextDividendPaymentAmount(Client client)
+    {
+        Column column = new Column("nextdivpmtamt", Messages.ColumnDividendsNextPaymentAmount, SWT.RIGHT, 60); //$NON-NLS-1$
+        column.setMenuLabel(Messages.ColumnDividendsNextPaymentAmount_MenuLabel);
+        column.setGroupLabel(Messages.GroupLabelDividends);
+        column.setVisible(false);
+        column.setLabelProvider(new ColumnLabelProvider()
+        {
+            @Override
+            public String getText(Object element)
+            {
+                Security sec = Adaptor.adapt(Security.class, element);
+                if (sec == null)
+                {
+                    return null; //
+                }
+                LocalDate now = LocalDate.now();
+                Money amount = getAmount(now, sec);
+
+                if (amount == null)
+                {
+                    return null; //
+                }
+                return Values.Money.format(amount, client.getBaseCurrency(), false);
+            }
+        });
+        column.setSorter(ColumnViewerSorter.create((o1, o2) -> {
+            Money m1 = getAmount(LocalDate.now(), (Security) o1);
+            Money m2 = getAmount(LocalDate.now(), (Security) o2);
+
+            if (m1 == null)
+                return m2 == null ? 0 : -1;
+            if (m2 == null)
+                return 1;
+
+            return m1.compareTo(m2);
+        }));
```

Comment:
We can use functions of the column viewer sorter. We must also adapt the object as the row objects is passed here as well.

```java
        column.setSorter(ColumnViewerSorter.create(element -> {
            Security security = Adaptor.adapt(Security.class, element);
            return security != null ? getAmount(now, security) : null;
        }));
```

Reply:
I've changed it accordingly